rules/coding-style/src/Rector/Throw_/AnnotateThrowablesRector.php

```diff
@@ -252,4 +338,55 @@ private function getThrowingStmtPhpDocInfo(Throw_ $throw): PhpDocInfo
 
         return $stmt->getAttribute(AttributeKey::PHP_DOC_INFO);
     }
+
+    private function getUsesFromReflectedFunction(ReflectionFunction $reflectionFunction): array
+    {
+        $functionNode = $this->parseFunction($reflectionFunction);
+        if ($functionNode === false) {
+            return [];
+        }
+
+        $uses = [];
+        foreach ($functionNode->stmts as $stmt) {
+            if (! $stmt instanceof Use_) {
+                continue;
+            }
+
+            $use = $stmt->uses[0];
+            if (! $use instanceof UseUse) {
+                continue;
+            }
+
+            $parts = $use->name->parts;
+            $uses[$parts[count($parts) - 1]] = implode('\\', $parts);
+        }
+
+        return $uses;
+    }
+
+    private function parseFunction(ReflectionFunction $reflectionFunction)
+    {
+        static $parser = null;
+        if ($parser === null) {
+            $parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+        }
+
+        $fileName = $reflectionFunction->getFileName();
+        if (! is_string($fileName)) {
+            return [];
+        }
+
+        $functionCode = FileSystem::read($fileName);
+        if (! is_string($functionCode)) {
+            return [];
+        }
+
+        $ast = $parser->parse($functionCode)[0];
+
+        if (! $ast instanceof Namespace_) {
+            return false;
+        }
+
+        return $ast;
```

Comment:
Better return `null` instead of `false`, so return type can by strict, e.g. `?Type`

Reply:
I don't know anymore... I'm refactoring now... I will inspect this after the refactoring.